cmake/common.cmake

```diff
@@ -748,3 +748,49 @@ function(add_picobench name)
 
   set_property(TEST ${name} PROPERTY LABELS benchmark)
 endfunction()
+
+# Perf tool executable
+set(SUBMITTER_DIR ${CCF_DIR}/tests/perf-system/submitter)
+
+add_executable(
+  submit ${SUBMITTER_DIR}/submit.cpp ${SUBMITTER_DIR}/handle_arguments.hpp
+         ${SUBMITTER_DIR}/parquet_data.hpp
+)
+
+add_library(stdcxxhttp_parser.host "${HTTP_PARSER_SOURCES}")
+
+set(CCFCRYPTO_SRC
+    ${CCF_DIR}/src/crypto/base64.cpp
+    ${CCF_DIR}/src/crypto/entropy.cpp
+    ${CCF_DIR}/src/crypto/hash.cpp
+    ${CCF_DIR}/src/crypto/sha256_hash.cpp
+    ${CCF_DIR}/src/crypto/symmetric_key.cpp
+    ${CCF_DIR}/src/crypto/key_pair.cpp
+    ${CCF_DIR}/src/crypto/rsa_key_pair.cpp
+    ${CCF_DIR}/src/crypto/verifier.cpp
+    ${CCF_DIR}/src/crypto/key_wrap.cpp
+    ${CCF_DIR}/src/crypto/hmac.cpp
+    ${CCF_DIR}/src/crypto/openssl/symmetric_key.cpp
+    ${CCF_DIR}/src/crypto/openssl/public_key.cpp
+    ${CCF_DIR}/src/crypto/openssl/key_pair.cpp
+    ${CCF_DIR}/src/crypto/openssl/hash.cpp
+    ${CCF_DIR}/src/crypto/openssl/rsa_public_key.cpp
+    ${CCF_DIR}/src/crypto/openssl/rsa_key_pair.cpp
+    ${CCF_DIR}/src/crypto/openssl/verifier.cpp
+)
+
+add_library(stdcxxccfcrypto.host STATIC "${CCFCRYPTO_SRC}")
+target_link_libraries(stdcxxccfcrypto.host PUBLIC crypto)
+target_link_libraries(stdcxxccfcrypto.host PUBLIC ssl)
+
+if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 9)
+  target_link_libraries(
+    submit PRIVATE stdcxxhttp_parser.host stdcxxccfcrypto.host arrow parquet
+  )
+
+else()
+  target_link_libraries(
+    submit PRIVATE stdcxxhttp_parser.host stdcxxccfcrypto.host arrow parquet
+                   c++fs
+  )
+endif()
```

Comment:
Repeated calls to `target_link_libraries` will append libraries. So it would be simpler to have a single call outside the conditional listing the common libraries, then:

```
if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 9)
  target_link_libraries(submit PRIVATE c++fs)
endif()
```

Reply:
Yes, great suggestion. I made my statement looking the following [line](https://github.com/microsoft/CCF/blob/94f53a0f8bd4035aa3bf541f4756e9f3538b3a62/cmake/common.cmake#L296) which does something similar and could benefit from the same suggestion. Happy to make the change in the line above, if people want me to.

Reply:
Ah ha! Yes, please fix that instance as well :)